ios/objc/src/TorchIValue.mm

```diff
@@ -0,0 +1,191 @@
+#include <vector>
+#import "TorchIValue.h"
+#import "TorchIValue+Internal.h"
+#import "TorchTensor.h"
+#import "TorchTensor+Internal.h"
+#import <Pytorch/Pytorch.h>
+
+#define IVALUE_TYPE(_) \
+    _(None) \
+    _(Bool) \
+    _(Int) \
+    _(Double) \
+    _(Tensor) \
+    _(BoolList) \
+    _(IntList) \
+    _(DoubleList)\
+    _(TensorList)
+
+#define IVALUE_SCALAR_TYPE(_) \
+    _(Bool, bool, bool) \
+    _(Int, int, int64_t) \
+    _(Double, double,double) \
+
+@implementation TorchIValue {
+    std::shared_ptr<at::IValue> _impl;
```

Comment:
IValue is a tiny type that is intended to be passed by value - there's no need in a shared_ptr (similarly to Tensor)